papis/format.py

```diff
@@ -153,11 +147,20 @@ def get_formater(name: Optional[str] = None) -> Formater:
 def format(fmt: str,
            doc: papis.document.DocumentLike,
            doc_key: str = "",
+           default: Optional[str] = None,
            additional: Optional[Dict[str, Any]] = None) -> str:
     """Format a string using the selected formater.
 
     This is the user-facing function that should be called when formating a
     string. The formaters should not be called directly.
     """
     formater = get_formater()
-    return formater.format(fmt, doc, doc_key=doc_key, additional=additional)
+    try:
+        return formater.format(fmt, doc, doc_key=doc_key, additional=additional)
+    except Exception as e:
+        if default is not None:
+            logger.debug("Failed to format according to '%s' - %s", fmt, e)
+            return default
+        else:
+            logger.warning("Failed to format according to '%s' - %s", fmt, e)
+            raise
```

Comment:
```suggestion
    except Exception as exc:     
        if default is not None:
        	logger.warning("Could not format string '%s'.", fmt, exc_info=exc)   
            return default
        else:
            raise FormatFailedError(fmt) from exc
```
What do you think about something like this? i.e. adding a `FormatFailedError` exception that would catch all the errors thrown here, so that callers would only have to deal with that.

Reply:
implemented.